tests/library/firefox/launcher.spec.ts

```diff
@@ -16,30 +16,39 @@
 
 import { playwrightTest as it, expect } from '../../config/browserTest';
 
-it('should pass firefox user preferences', async ({ browserType, mode }) => {
+it('should pass firefox user preferences', async ({ browserType, mode, server }) => {
   it.skip(mode.startsWith('service'));
+
   const browser = await browserType.launch({
     firefoxUserPrefs: {
-      'network.proxy.type': 1,
-      'network.proxy.http': '127.0.0.1',
-      'network.proxy.http_port': 3333,
+      'javascript.enabled': false,
     }
   });
+
+  server.setRoute('/empty.html', async (req, res) => {
+    res.end(`<script>window.testStatus = 'SUCCESS';</script>`);
+  });
+
   const page = await browser.newPage();
-  const error = await page.goto('http://example.com').catch(e => e);
-  expect(error.message).toContain('NS_ERROR_PROXY_CONNECTION_REFUSED');
+  await page.goto(server.EMPTY_PAGE);
+
+  expect(await page.evaluate(() => window['testStatus'])).toBe(undefined);
```

Comment:
This line will succeed if page.goto does not work, or if server.setRoute does not work. Let's add a DOM node and check that the DOM is present.

Reply:
That is true! Thanks for pointing that out - it makes sense.

After some testing it turns out that disabling JavaScript globally in the browser has a [side-effect in Firefox](https://bugzilla.mozilla.org/show_bug.cgi?id=1942990). As such I would suggest to revert the above changes and still continue to use the proxy solution.

Hereby we still have to make 2 changes:

1. Lets use `https://example.com` as remote site which not triggering different behavior in Nightly and DevEdition builds due to the enabled HTTPS-First feature.
2. Update the preferences to set the relevant SSL proxy.